Source/DafnyStandardLibraries/src/Std/DateTime/Duration.dfy

```diff
@@ -0,0 +1,365 @@
+include "DateTimeConstant.dfy"
+
+module Std.Duration {
+  import opened DateTimeConstant
+  import opened Strings
+  import opened Collections.Seq
+  import opened BoundedInts
+
+  datatype Duration = Duration(
```

Comment:
Just a thought: I was thinking of suggesting we define an `Instant` type to represent epoch values, but then realized that `Duration` can serve that purpose as well. After all, specifying a time via a number always has to ultimately be a Duration since a fixed point like the epoch!

Given that, it might be nice to have a `LocalDataTime.ToEpochTime(): Duration` function